scheduler/remote_server.py

```diff
@@ -76,51 +76,109 @@ def get_cmap(n, name='hsv'):
+            midnight = datetime.datetime.combine(event['start'] + td, datetime.datetime.min.time())
+
+            first_dur = midnight - event['start']
+            second_dur = event['end'] - midnight
+
+            # handle the new event first
+            new_event['start'] = midnight
+            new_event['duration'] = second_dur
+            new_event['end'] = event['end']
+
+            # now handle the old event
+            event['duration'] = first_dur
+            event['end'] = midnight
+
+            return [event] + split_event(new_event)
+
+
+    # make random colors
     cmap = get_cmap(len(timeline_dict.items()))
     colors = [cmap(i) for i in range(len(timeline_dict.items()))]
     random.shuffle(colors)
 
+    # put the colors in and create event records with only start and end times,
+    # durations, colors, and labels
     for i, (_, events) in enumerate(timeline_dict.items()):
-        event_times = []
-        event_label = ""
-
-        color = colors[i]
-        colors_for_line = [color] * len(events)
         for event in events:
-            time_start = mdates.date2num(event['time'])
+            event['color'] = colors[i]
 
-            if event['duration'] == '-':
-                td = get_next_month(event['time']) - event['time']
-            else:
-                td = datetime.timedelta(minutes=int(event['duration']))
+    for i, event in enumerate(timeline_list):
+        event_item = dict()
 
-            time_end = td.total_seconds()/(24 * 60 * 60)
-            event_times.append((time_start, time_end))
-            event_label = event['experiment']
+        event_item['color'] = event['color']
+        event_item['label'] = event['experiment']
 
-            if i == 0:
-                first_date = event['time']
-            if i == len(timeline_dict.items()) - 1:
-                last_date = event['time'] + td
+        event_item['start'] = event['time']
 
-        event_labels.append(event_label)
-        ax.broken_barh(event_times, ((i+1)*10, 4), facecolors=colors_for_line)
+        if event['duration'] == '-':
+            td = get_next_month(event['time']) - event['time']
+        else:
+            td = datetime.timedelta(minutes=int(event['duration']))
+
+        event_item['end'] = event_item['start'] + td
+        event_item['duration'] = td
+
+        timeline_list[i] = event_item
+
+        if i == 0:
+            first_date = event['time'].date()
+        if i == len(timeline_list) - 1:
+            last_date = event['time'] + td
+            plot_last = (last_date + datetime.timedelta(hours=12)).date()
+
+    # loop through events again, splitting them where necessary
+    for event in timeline_list:
+        event_list += split_event(event)
+
+    for event in event_list:
+        day_offset = event['start'].date() - first_date
+        start = event['start'] - day_offset
+        ax.barh(event['start'].date(), event['duration'], 0.16, left=start, color=event['color'], align='edge')
```

Comment:
Do you think 16 is a good bar width if you have like 30+ days being plotted? would this maybe be better if it made a series of plots with like 10 or so days max on each?

Reply:
This is probably a good idea. Things would be much less busy.

Reply:
I think I'd prefer the whole month in one plot because that's how we typically schedule and I think that plot could be useful to show others (externally) the month breakdown. As long as the dpi of the plot is sufficient so that we can read the experiment names.

Reply:
I think we can start and see what it looks like, but I think it will look crowded.